File: src/common/models/dto/UserDto.ts
import { Gender } from "src/common/enums/Gender";
import { Roles } from "src/common/enums/Roles";

export interface UserDto {
  id: number;
  email: string;
  firstName: string;
  lastName: string;
  roleId: Roles;
  pesel: number;
  hiringDate: Date;
  numberOfYearsWorkedOnHiringDate: number;
  genderId: Gender;
}

File: TwojUrlop.DomainModel/Common/Models/UserDto.cs
using TwojUrlop.Common.Enums;

namespace TwojUrlop.DomainModel.Common;
public class UserDto
{
    public required string Email { get; set; }
    public string FirstName { get; set; } = default!;
    public string LastName { get; set; } = default!;
    public int RoleId { get; set; }
    public int PESEL { get; set; }
    public DateTime HiringDate { get; set; }
    public int NumberOfYearsWorkedOnHiringDate { get; set; }
    public int GenderId { get; set; }
}